dsgrid/dataset/pivoted_table.py

```diff
@@ -0,0 +1,197 @@
+import copy
+import logging
+from typing import List
+
+import pyspark.sql.functions as F
+
+from dsgrid.dimension.base_models import DimensionType
+from dsgrid.exceptions import DSGInvalidParameter
+from dsgrid.query.models import AggregationModel
+from dsgrid.query.query_context import QueryContext
+from dsgrid.utils.dataset import map_and_reduce_pivoted_dimension
+from dsgrid.utils.spark import get_unique_values
+from .table_format_handler_base import TableFormatHandlerBase
+
+
+logger = logging.getLogger(__name__)
+
+
+class PivotedTableHandler(TableFormatHandlerBase):
+    """Implements behavior for tables stored in pivoted format."""
+
+    def __init__(self, project_config, dataset_id=None):
+        super().__init__(project_config, dataset_id=dataset_id)
+
+    def add_columns(self, df, query_columns, context: QueryContext, aggregation_allowed):
+        columns = set(df.columns)
+        base_query_names = self.project_config.get_base_dimension_query_names()
+        for column in query_columns:
+            query_name = column.dimension_query_name
+            if query_name in base_query_names:
```

Comment:
Still need these checks? Feels like you want to see that `query_name` is in both `base_query_names` and `columns`.